components/raftstore/src/store/async_io/write.rs

```diff
@@ -0,0 +1,635 @@
+                        "failed to send msg to other peer in async-writer";
+                        "region_id" => task.region_id,
+                        "peer_id" => task.peer_id,
+                        "target_peer_id" => to_peer_id,
+                        "target_store_id" => to_store_id,
+                        "err" => ?e,
+                        "error_code" => %e.error_code(),
+                    );
+                    self.message_metrics.add(msg_type, false);
+                    // If this msg is snapshot, it is unnecessary to send snapshot
+                    // status to this peer because it has already become follower.
+                    // (otherwise the snapshot msg should be sent in store thread other than here)
+                    // Also, the follower don't need flow control, so don't send
+                    // unreachable msg here.
+                } else {
+                    self.message_metrics.add(msg_type, true);
+                }
+            }
+        }
+        if self.trans.need_flush() {
+            self.trans.flush();
+            self.message_metrics.flush();
+        }
+        let now2 = Instant::now();
+        STORE_WRITE_SEND_DURATION_HISTOGRAM
+            .observe(duration_to_sec(now2.saturating_duration_since(now)));
+
+        for (region_id, (peer_id, ready_number)) in &self.batch.readies {
+            self.notifier
+                .notify_persisted(*region_id, *peer_id, *ready_number, now2);
+        }
+        STORE_WRITE_CALLBACK_DURATION_HISTOGRAM.observe(duration_to_sec(now2.saturating_elapsed()));
+
+        self.batch.clear();
+
+        fail_point!("raft_after_save");
+    }
+}
+
+pub struct StoreWriters<EK, ER>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+{
+    writers: Vec<Sender<WriteMsg<EK, ER>>>,
+    handlers: Vec<JoinHandle<()>>,
+}
+
+impl<EK, ER> StoreWriters<EK, ER>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+{
+    pub fn new() -> Self {
+        Self {
+            writers: vec![],
+            handlers: vec![],
+        }
+    }
+
+    pub fn senders(&self) -> &Vec<Sender<WriteMsg<EK, ER>>> {
+        &self.writers
+    }
+
+    pub fn spawn<T: Transport + 'static, N: Notifier>(
+        &mut self,
+        store_id: u64,
+        engines: &Engines<EK, ER>,
+        notifier: &N,
+        trans: &T,
+        cfg: &Arc<VersionTrack<Config>>,
+    ) -> Result<()> {
+        let pool_size = cfg.value().store_io_pool_size;
+        for i in 0..pool_size {
+            let tag = format!("store-writer-{}", i);
+            let (tx, rx) = bounded(cfg.value().store_io_notify_capacity);
+            let mut worker = Worker::new(
+                store_id,
+                tag.clone(),
+                engines.clone(),
+                rx,
+                notifier.clone(),
+                trans.clone(),
+                cfg,
+            );
+            let t = thread::Builder::new().name(thd_name!(tag)).spawn(move || {
+                worker.run();
```

Comment:
How about adding some logs when a worker is started or stopped.

Reply:
Done